sphinx/tutorial/rose/furthertopics/date-time-manipulation.rst

```diff
@@ -2,18 +2,117 @@ Date and Time Manipulation
+-------------------------
 
-``rose date``
--------------
+The ``rose date`` command provides functionality for:
 
-See :ref:`rose-date-label`
+* Parsing and formatting datetimes e.g:
+
+  .. code-block:: console
+
+     $ rose date '12-31-2000' --parse-format='%m-%d-%Y' 
+     12-31-2000
+     $ rose date '12-31-2000' --parse-format='%m-%d-%Y' --format='DD-MM-CCYY'
+     31-12-2000
+
+* Adding offsets to datetimes e.g:
+
+  .. code-block:: console
+
+     $ rose date '2000-01-01T0000Z' --offset '+P1M'
+     2000-02-01T0000Z
+
+* Calculating the duration between two datetimes e.g:
+
+  .. code-block:: console
+
+     $ rose date '2000' '2001'
+     P366D
+
+See the :ref:`command-rose-date` command reference for more information.
+
+
+Using ``rose date`` In A Suite
+------------------------------
+
+In datetime cycling suites ``rose date`` can work with the cyclepoint using
+the ``CYLC_TASK_CYCLE_POINT`` environment variable:
+
+.. code-block:: cylc
+
+   [runtime]
+       [[hello_america]]
+           script = rose date $CYLC_TASK_CYCLE_POINT --format='MM-DD-CCYY'
+
+Alternatively, if you are providing the standard rose task environment using
+:ref:`command-rose-task-env` then :ref:`command-rose-date` can use the ``-c``
+option to pick up the cycle point:
+
+.. code-block:: cylc
+
+   [runtime]
+       [[hello_america]]
+           env-script = eval $(rose task-env)
+           script = rose date -c --format='MM-DD-CCYY'
+
+
+The ``ROSE_DATAC`` Environment Variable
+---------------------------------------
+
+There are two locations where task output is likely to be located:
+
+The work directory
+   Each task is executed within its :term:`work directory` which is located in:
+
+   .. code-block:: sub
+
+      <run directory>/work/<cycle>/<task-name>
+
+   The path a task's work directory can be obtained from the
+   ``CYLC_TASK_WORK_DIR`` environment variable.
+
+The share directory
+   The :term:`share directory` serves the purpose of providing a storage place
+   for any files which need to be shared between different tasks.
+   
+   Within the share directory data is typically stored within cycle
+   subdirectories i.e:
+
+   .. code-block:: sub
+
+      <run directory>/share/<cycle>
+
+   These are called the ``share/cycle`` directories.
+
+   The path to the root of the share directory is provided by the
+   ``CYLC_SUITE_SHARE_DIR`` environment variable so the path to the cycle
+   subdirectory would be::
+
+      "$CYLC_SUITE_SHARE_DIR/$CYLC_SUITE_CYCLE_POINT"
+
+The ``rose task-env`` command provides the environment variable
+``ROSE_DATAC`` which is a more convenient way to obtain the path of the
+``share/cycle`` directory.
+
+To get the path to a previous (or a future) ``share/cycle`` directory we can
+provide an offset to ``rose task-env`` e.g::
+
+   rose task-env --cycle-offset=PT1H
+
+The path is then made available as the ``ROSE_DATACPT1H`` environment variable.
+
+.. TODO - Write a short practical using ROSE_DATAC and rose-date.
```

Comment:
I have been having a think about ideas for this short practical, as suggested, and have thought of a few possible scenarios that could work and hopefully not seem too contrived. They may be too complex so feel free to simplify if you wanted to use something along these lines.

1. _Birthday card scenario_: a (largeish) company wants to send a birthday card to all its employees on their birthday. They know all employee DOB so firstly as a daily task parse the birthdays to see which if any birthdays have a month and day matching those of the current date plus two days as an offset (so they have time to buy and sign the cards to deliver on the birthdays themselves). So it would have a graph similar to "check_birthdays => buy_cards => CEO_sign_cards => deliver_cards_to_desks" with 'check_birthdays' recurring daily. Then as a possible add-on part of the tutorial at the end, could say the company wants to create a custom "Happy Xth birthday" message in the card, so can calculate the person's new age in years to put in the message via ``$ rose date <birthday> <current day with 2 days offset>``. So that can cover all of parsing, adding offsets and calculating durations.

2. Or a similar scenario but with retirement cards/gifts, assuming consistent retirement age?

3. _An 'international' scenario_: has a suite where different tasks need calculations to convert to different time zones  (I am aware there is the UTC Mode for altering the time zone if required for the entire suite system clock but if individual tasks had to do various time-zone calculations). For example, maybe, a commercial aeroplane company providing long-haul flights which make a few intermediate stops before reaching a final terminal airport want flight crews to update passengers on the weather nowcast at each intermediate stop so have a system sending the required nowcast to the crew of each flight which needs to deal with timezone in those countries via offset in order to obtain the correct forecast information.